### DISCUSSION TITLE
How can I set the data directory?

### DISCUSSION THREAD

**Discussion by evidwong** [rabbitmq/rabbitmq-server#2696] [Other] [answered]
Use the configure file /etc/rabbitmq/rabbitmq-env.conf for version 3.8.9 in centos 7.5, it call an error "RABBITMQ_MNESIA_BASE:command not found" when i starting rabbitmq-server.How can I set the data directory in configure file?

**Comment by michaelklishin** [comment, marked as answer]
I'm afraid we cannot suggest much with a single line message. We do not guess in this community. Please help others help you by sharing more relevant information:

 * The *exact* `rabbitmq-env.conf` contents
 * How *exactly* RabbitMQ was installed and is started
 * Complete standard output and error streams of the node, plus logs at debug level

Without the above information we will not be able to help you.

All you have to do is to set the [`RABBITMQ_MNESIA_DIR `](https://www.rabbitmq.com/relocate.html) variable value in a way that will expose the new value to the node (`rabbitmq-env.conf` is one way of doing it). The message suggests that `RABBITMQ_MNESIA_BASE` is not evaluated as an environment variable but rather as an executable invocation, so `rabbitmq-env.conf` may be off.

Note that `RABBITMQ_MNESIA_DIR` is sufficient to relocate node's data. `RABBITMQ_MNESIA_BASE` is needed only when multiple nodes are started on the same host, which is only relevant in certain development and QA environments and *never* recommended for production.
